faq.md

```diff
@@ -8,136 +8,313 @@ permalink: /faq/
+the framework and widgets. The underlying graphics framework and the Dart
+virtual machine are implemented in C/C++.
+
+### Why did Flutter choose to use Dart?
+
+The primary criteria we used to pick a programming language were the following:
+
+* _Developer productivity_. One of Flutter's main value propositions is that it
+  saves engineering resources by letting developers create apps for both iOS and
+  Android with the same codebase. Using a highly productive language
+  accelerates developers further and makes Flutter more attractive.
+
+* _Object-orientation_. For Flutter, we want a language that's suited to
+  Flutter's problem domain: creating visual user experiences. The industry has
+  multiple decades of experience building user interface frameworks in
+  object-oriented languages. While we could use a non-object-oriented language,
+  this would mean reinventing the wheel to solve several hard problems.
+
+* _Predictable, high performance_. With Flutter, we want to empower developers
+  to create fast, fluid user experiences. In order to achieve that, we need to
+  be able to run a significant amount of end-developer code during every
+  animation frame. That means we need a language that both delivers high
+  performance and delivers predictable performance, without periodic
+  pauses that would cause dropped frames.
 
-## What kind of app performance can I expect?
+* _Fast allocation_. The Flutter framework uses a functional-reactive style
+  programming model, whose performance depends heavily on the underlying memory
+  allocator efficiently handling small, short-lived allocations. The
+  functional-reactive style was developed in languages with this property and
+  does not work efficiently in languages that lack this facility.
+
+Dart scores highly on all four dimensions. In addition, we have the opportunity
+to work closely with the Dart community, which is actively investing resources
+in improving Dart for use in Flutter. For example, when we
+adopted Dart, the language did not have an ahead-of-time toolchain for producing
+native binaries, which is instrumental in achieving predictable, high
+performance, but now the language does because the Dart team built it for
+Flutter. Similarly, the Dart VM has previously been optimized for throughput
+but the team is now optimizing the VM for latency, which is more important for
+Flutter's workload.
+
+### Can Flutter run any Dart code?
+
+Flutter should be able to run most Dart code that does not import (transitively,
+or directly) dart:mirrors or dart:html.
+
+### How big is the Flutter engine?
+
+In November 2015, we measured the size of a minimal Flutter app, bundled as
+an APK, to be approximately 8MB. For this simple app that used Material Design
+widgets, the core engine is approximately 5MB, the framework + app code is
+approximately 400kb, necessary Java code is 330k, and there is approximately
+2.5MB of ICU data. We are working to get this smaller.
```

Comment:
"We are working to get this smaller." is this important to say?  It seems apologetic.